### PR TITLE
[ray] disable flaky check

### PR DESCRIPTION
**Pull request by richardliaw**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs?
- [x] Did you write any tests to validate this change?  
- [x] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes https://github.com/horovod/horovod/issues/2438

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Fixed
 
 - Fixed `local_rank` support for Ray. ([#2596](https://github.com/horovod/horovod/pull/2596))
+- Disabled flaky resource check for Ray `test_colocator`. ([#2628](https://github.com/horovod/horovod/pull/2628))
```
Comment (EnricoMi) [inline CHANGELOG.md:22]: Are changes in the test infrastructure worth listing in the CHANGELOG? No user impact. @tgaddair?

Comment (tgaddair) [reply CHANGELOG.md:22]: I agree, we can leave it out of the changelog.

Comment (richardliaw) [reply CHANGELOG.md:22]:
```suggestion
```

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Comment by EnricoMi** [comment]
Oh, you need to rebase and sign your commit again, the [DCO check](https://github.com/horovod/horovod/pull/2628/checks?check_run_id=1788364381) is complaining.

**Comment by richardliaw** [comment]
@EnricoMi thanks for the ping -- done!

**Pull request merged**